### ISSUE TITLE
Heartbeat failed unknown method UpsertDatabaseService for service proto.AuthService

### ISSUE THREAD

**Issue by mdaniel** [gravitational/teleport#19919]
Expected behavior:

`helm install teleport-kube-agent --version 11.1.4` installs a working teleport-agent

Current behavior:

While the helm install does complete, the Pod logs contain the following warning which appears to be preventing it from becoming Ready and thus from scaling the STS up to full replicas count

> 2023-01-05T19:31:58Z WARN [DB:SERVIC] Heartbeat failed unknown method UpsertDatabaseService for service proto.AuthService. srv/heartbeat.go:282

When running with `--debug` there are a bunch of attempts to `iam:GetRolePolicy` and the extremely alarming [`rds:ModifyDBCluster`](https://github.com/gravitational/teleport/blob/v11.1.4/lib/srv/db/cloud/aws.go#L151-L155) but they're logged only at debug and, yes, we have never and will never allow teleport-agent to mutate RDS clusters nor update IAM resources

_update:_

So I tried with 11.1.3 and while it gave off a different error on startup, its Pods came Ready so it seems this is a regression for 11.1.4 specifically

Bug details:
- Teleport version 11.1.4
- Recreation steps
- Debug logs

**Comment by zmb3** [comment]
Thanks for filing this issue.

Looks like your agent is running 11.1.4 but your auth server is on an older version. This should work per our compatibility guidelines but doesn't due to some faulty logic in #19570.

We'll get a fix out as soon as possible. In the meantime, downgrading the agent is probably the best remediation.

**Comment by mdaniel** [comment]
Ok, thanks for checking into that. Do you want me to close this, or leave it open?

**Comment by zmb3** [comment]
Open is great, thanks!

**Comment by marcoandredinis** [comment]
We ended up reverting those developments.
Next release should work as expected, sorry for the trouble

**Comment by mdaniel** [comment]
Did you pull [v11.1.3](https://github.com/gravitational/teleport/releases/v11.1.3)? I ask because as I said in my description, it's the one I rolled back to when .4 did not shake out and now it's out in the helm repo
```yaml
  - apiVersion: v2
    appVersion: 11.1.3
    created: "2022-12-25T23:14:58.514325129Z"
    description: Teleport provides a secure SSH and Kubernetes remote access solution
      that doesn't get in the way.
    digest: e5c786df78787ca1abb3ea4b29adcc4b6acfd0fac96746a351bb16e1ba51f94f
    icon: https://goteleport.com/images/logos/logo-teleport-square.svg
    keywords:
    - Teleport
    name: teleport-kube-agent
    urls:
    - teleport-kube-agent-11.1.3.tgz
```
and there are 11.1.3 images:
```console
$ helm template --debug --version 11.1.3 fred teleport/teleport-kube-agent --set proxyAddr=foo --set kubeClusterName=bar | grep image
        image: "public.ecr.aws/gravitational/teleport:11.1.3"
```
but if the release is bogus in some way, I now have to redo my internal PRs that reference that version

I can't tell if I accidentally forgot to check there was such a release, meaning I just rolled down one and "it worked," or the tag was pulled

**Comment by zmb3** [comment]
11.1.3 is totally fine. This issue was introduced in 11.1.4 and is fixed in 11.2.0 which is publishing now.
